src/subdomains/generic/kyc/services/kyc.service.ts

```diff
@@ -266,16 +266,24 @@ export class KycService {
     return KycStepMapper.toKycResult(kycStep);
   }
 
-  async updateUserData(
+  async updateKycStep(
     kycHash: string,
     stepId: number,
     data: Partial<UserData>,
     requiresInternalReview: boolean,
+    updateUserData: boolean,
   ): Promise<KycResultDto> {
     let user = await this.getUser(kycHash);
     const kycStep = user.getPendingStepOrThrow(stepId);
 
-    user = await this.userDataService.updateUserDataInternal(user, data);
+    if (data.nationality) {
+      const nationality = await this.countryService.getCountry(data.nationality.id);
+      if (!nationality) throw new BadRequestException('Nationality not found');
+
+      data.nationality.symbol = nationality.symbol;
```

Comment:
@davidleomay wdyt, is it better to have nationality: {id: number, symbol: string} as kycStepResult or just nationality: string with the symbol instead of the id? 🤔 If we decide to split into two methods we could do that or do we need the countryId in the stepResult, too?

Reply:
I think it's ok to just store the data as it comes, the ID should be enough, don't you think?